src/lattices/abstract.jl

```diff
@@ -0,0 +1,66 @@
+"""
+Abstract definition of a lattice.
+"""
+abstract type AbstractLattice end
+
+# AbstractLattice interface: mandatory
+# TODO: This needs to be updated. There must be a general way to access sites and bonds.
+"""
+    length(l::AbstractLattice)
+
+Number of lattice sites.
+"""
+@inline Base.length(l::AbstractLattice) = error("Lattice $(typeof(l)) doesn't implement `length`.")
+
+
+
+
+
+
+# AbstractLattice interface: optional
+"""
+    neighbors(l::AbstractLattice, drop_repeats::Val{true})
+    neighbors(l::AbstractLattice, drop_repeats::Val{false})
+
+Returns an iterator over bonds, given as tuples (source index, target index). If
+`drop_repeats = Val(false)` bonds are assumed to be directed, i.e. both
```

Comment:
`directed` seems much better than `drop_repeats` to me.

Reply:
I agree